homeassistant/components/mqtt/client.py

```diff
@@ -494,6 +581,24 @@ def _async_untrack_subscription(self, subscription: Subscription) -> None:
         except (KeyError, ValueError) as ex:
             raise HomeAssistantError("Can't remove subscription twice") from ex
 
+    async def _async_queue_subscriptions(
+        self, subscriptions: Iterable[tuple[str, int]], queue_only: bool = False
+    ) -> None:
+        """Queue requested subscriptions."""
+        async with self._pending_subscriptions_lock:
+            for subscription in subscriptions:
+                topic, qos = subscription
+                old_qos: int = self._max_qos.setdefault(topic, qos) or 0
+                new_qos = max([old_qos, qos])
+                if new_qos > old_qos:
+                    self._max_qos[topic] = new_qos
+                if self._pending_subscriptions.setdefault(topic, new_qos) < new_qos:
+                    # update the qos if the existing value has a lower qos
+                    self._pending_subscriptions[topic] = new_qos
```

Comment:
This seems overly complicated, can't it just be simplified like this?

```suggestion
                max_qos = max(qos, self._max_qos.setdefault(topic, qos))
                self._max_qos[topic] = max_qos
                self._pending_subscriptions[topic] = max_qos
```